.github/workflows/test.yml

```diff
@@ -0,0 +1,43 @@
+on:
+  pull_request:
+    branches:
+      - main
+      - master
+  push:
+    branches:
+      - main
+      - master
+  release:
+    types: [published]
+
+env:
+  GOFLAGS: "-mod=vendor"
+
+jobs:
+  test:
+    name: Test and Publish
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v4
+
+      - uses: actions/setup-go@v5
+        with:
+          go-version: '1.16'
+
+      - name: Setup
+        run: |
+          make tools
+          go mod vendor
+
+      - name: Code Lint
+        run:  make lint
+
+      - name: Code UnitTest
+        run: make test
+
+      - name: Deploy
```

Comment:
what about the GPG_PRIVATE_KEY private key part from the travis config?

Reply:
Aha. I went ahead and moved to some publicly available actions for deploy, based on the docs for what we were using recommending it : https://goreleaser.com/ci/actions/?h=gpg#signing